homeassistant/components/qnap_qsw/coordinator.py

```diff
@@ -41,3 +42,29 @@ async def _async_update_data(self) -> dict[str, Any]:
             except QswError as error:
                 raise UpdateFailed(error) from error
             return self.qsw.data()
+
+
+class QswFirmwareCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching firmware data from the QNAP QSW device."""
+
+    def __init__(self, hass: HomeAssistant, qsw: QnapQswApi) -> None:
+        """Initialize."""
+        self.qsw = qsw
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=FW_SCAN_INTERVAL,
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Update firmware data via library."""
+        async with async_timeout.timeout(QSW_TIMEOUT_SEC):
+            try:
+                await self.qsw.check_firmware()
+            except APIError as error:
+                _LOGGER.warning(error)
```

Comment:
Should this also raise update failed?

Reply:
While I was testing this QNAP servers went down (or the update file fetched by the switch was malformed) and it prevented the whole integration from starting.
I changed this so the integration would start even if the firmware update check fails. This way the other platforms will still work even though the firmware update can't be checked.

Reply:
It's better to still raise `UpdateFailed` here but catch `ConfigEntryNotReady` when calling `coordinator.async_config_entry_first_refresh` for this coordinator.

Reply:
> It's better to still raise `UpdateFailed` here but catch `ConfigEntryNotReady` when calling `coordinator.async_config_entry_first_refresh` for this coordinator.

I will address this in a new PR.
Thanks @MartinHjelmare

Reply:
Resolved by https://github.com/home-assistant/core/pull/74288